atom/browser/api/atom_api_window.cc

```diff
@@ -191,6 +191,12 @@ void Window::OnWindowClosed() {
       FROM_HERE, GetDestroyClosure());
 }
 
+#if defined(OS_WIN)
+void Window::OnWindowEndSession() {
+  Emit("endsession");
```

Comment:
I think this event name should be `session-end` to match the other "end" events we currently have, `drag-end` and `scroll-touch-end`